lib/Driver/ToolChain.cpp

```diff
@@ -62,6 +62,27 @@ ToolChain::JobContext::getTemporaryFilePath(const llvm::Twine &name,
   return C.getArgs().MakeArgString(buffer.str());
 }
 
+ToolChain::ResponseFileInfo ToolChain::getResponseFileInfoIfNeeded(
+    const Compilation &C, const char *executablePath,
+    const ToolChain::InvocationInfo &invocationInfo,
+    const ToolChain::JobContext &context) const {
+  const bool forceResponseFiles =
+      C.getArgs().hasArg(options::OPT_driver_force_response_files);
+  assert((invocationInfo.allowsResponseFiles || !forceResponseFiles) &&
+         "Cannot force response file if platform does not allow it");
+
+  if (forceResponseFiles || (invocationInfo.allowsResponseFiles &&
+                             !llvm::sys::commandLineFitsWithinSystemLimits(
+                                 executablePath, invocationInfo.Arguments))) {
+    const char *responseFilePath =
+        context.getTemporaryFilePath("arguments", "resp");
+    const char *responseFileArg =
+        C.getArgs().MakeArgString(Twine("@") + responseFilePath);
+    return {responseFilePath, responseFileArg};
+  }
+  return {nullptr, nullptr};
```

Comment:
`return None` assuming you use `Optional<...>`

Reply:
Done.